Custom cache-control header seems to be overridden when using a file as mock body

A test of ours broke when trying to update to 16.3.0 from 16.1.3 (think), and the root cause seems to be using custom headers, which the test relies on. Setting a custom header works fine when the mock is using a string as body, but when the body is a file, the cache-control header is reverted to a default value of 1000. 

(I know I have commit rights to the repo but ran into all sorts of setup problems (tests failing out of the box both unit and integration) when trying get the repo up and running to create a failing test case. So reproducing usage code is the best I can do ATM)

I think it's this line that goes into effect, even though I'm unsure as to why it happens for files and not strings as mock body. 
https://github.com/popeindustries/dvlp/blob/main/src/mock/index.js#L784
```
return maxAge && maxAge[1] ? parseInt(maxAge[1], 10) * 1000 : 0;
```
Shortcutting the logic to 
```
return maxAge;
```
I get the specified cache-control header value, but that probably breaks some other use case I'm not aware of.


```
import { expect } from 'chai';
import { testServer, TestServer } from 'dvlp/dvlp-test.js';
import ky from 'ky';

const url = `http://test.url.com/req`;

describe.only('foo', function () {
  let server: TestServer;

  before(async () => {
    testServer.disableNetwork();
    server = await testServer({ autorespond: false, latency: 25, port: 3001 });
  });

  after(async () => {
    testServer.enableNetwork();
    await server.destroy();
  });

  it('should work', async function () {
    const headers = { 'Cache-Control': 'private, max-age=1' };
    server.mockResponse(url, { headers, body: '{ "done": true }' });
    const response = await loadUrl(url);
    expect(response.headers.get('Cache-Control')).to.equal('private, max-age=1');
  });

  it('should work too', async function () {
    const headers = { 'Cache-Control': 'private, max-age=1' };
    server.mockResponse(url, { headers, body: './a/file.json' });
    const response = await loadUrl(url);
    expect(response.headers.get('Cache-Control')).to.equal('private, max-age=1');
  });
});

async function loadUrl(url: string) {
  return ky.get(url);
}
```

gives a result of
```
foo
    ✔ should work (52ms)
    1) should work too


  1 passing (112ms)
  1 failing

  1) foo
       should work too:

      AssertionError: expected 'public, max-age=1000' to equal 'private, max-age=1'
      + expected - actual

      -public, max-age=1000
      +private, max-age=1
```

--- a/package.json
+++ b/package.json
@@ -33,7 +33,7 @@
   },
   "dependencies": {
     "commander": "^12.0.0",
-    "esbuild": "~0.20.1"
+    "esbuild": "~0.20.2"
   },
   "devDependencies": {
     "@fidm/x509": "^1.2.1",
@@ -41,19 +41,19 @@
     "@types/glob": "^8.1.0",
     "@types/mime": "^3.0.4",
     "@types/mocha": "^10.0.6",
-    "@types/node": "^20.11.27",
+    "@types/node": "^20.12.7",
     "@types/platform": "^1.3.6",
     "@types/send": "^0.17.4",
-    "@typescript-eslint/eslint-plugin": "^7.2.0",
-    "@typescript-eslint/parser": "^7.2.0",
+    "@typescript-eslint/eslint-plugin": "^7.6.0",
+    "@typescript-eslint/parser": "^7.6.0",
     "chai": "^5.1.0",
     "chalk": "^5.3.0",
     "chokidar": "^3.6.0",
     "cjs-module-lexer": "^1.2.3",
     "cross-env": "^7.0.3",
     "debug": "^4.3.4",
-    "electron": "29.1.4",
-    "es-module-lexer": "~1.4.1",
+    "electron": "29.3.0",
+    "es-module-lexer": "~1.5.0",
     "eslint": "^8.57.0",
     "eslint-config-prettier": "^9.1.0",
     "eventsource": "^2.0.2",
@@ -63,15 +63,15 @@
     "husky": "^9.0.11",
     "lint-staged": "^15.2.2",
     "lit-html": "^3.1.2",
-    "mocha": "^10.3.0",
-    "path-to-regexp": "^6.2.1",
+    "mocha": "^10.4.0",
+    "path-to-regexp": "^6.2.2",
     "permessage-deflate": "~0.1.7",
     "platform": "^1.3.6",
     "prettier": "^3.2.5",
     "react": "^18.2.0",
     "resolve.exports": "^2.0.2",
-    "terser": "^5.29.2",
-    "typescript": "^5.4.2"
+    "terser": "^5.30.3",
+    "typescript": "^5.4.5"
   },
   "engines": {
     "node": ">=18",

--- a/src/mock/index.js
+++ b/src/mock/index.js
@@ -18,8 +18,6 @@ import { Metrics } from '../utils/metrics.js';
 import path from 'node:path';
 import { send } from '../utils/send.js';
 
-const RE_MAX_AGE = /max-age=(\d+)/;
-
 const debug = Debug('dvlp:mock');
 const mockClient =
   // @ts-ignore
@@ -273,11 +271,6 @@ export class Mocks {
 
     switch (type) {
       case 'file': {
-        const maxAge =
-          getMaxAgeFromHeaders(
-            normaliseHeaderKeys(headers, ['Cache-Control']),
-          ) || Number(config.maxAge);
-
         // Set custom headers
         for (const header in headers) {
           res.setHeader(header, headers[header]);
@@ -285,7 +278,9 @@ export class Mocks {
         if (!res.hasHeader('Access-Control-Allow-Origin')) {
           res.setHeader('Access-Control-Allow-Origin', '*');
         }
-        res.setHeader('Cache-Control', `public, max-age=${maxAge}`);
+        if (!res.hasHeader('Cache-Control')) {
+          res.setHeader('Cache-Control', `public, max-age=${config.maxAge}`);
+        }
 
         send(
           // @ts-expect-error - body is path to file (relative to mock file)
@@ -764,22 +759,4 @@ function normaliseHeaderKeys(headers, keys) {
   }
 
   return normalisedHeaders;
-}
-
-/**
- * Retrieve max-age in ms from "headers" Cache-Control string
- *
- * @param { { [key: string]: string } } headers
- * @returns { number }
- */
-function getMaxAgeFromHeaders(headers) {
-  const cacheControl = headers['Cache-Control'];
-
-  if (!cacheControl) {
-    return 0;
-  }
-
-  const maxAge = RE_MAX_AGE.exec(cacheControl);
-
-  return maxAge && maxAge[1] ? parseInt(maxAge[1], 10) * 1000 : 0;
 }

--- a/test/unit/test-server-test.js
+++ b/test/unit/test-server-test.js
@@ -137,7 +137,7 @@ describe('testServer', () => {
     testServer.disableNetwork(false);
   });
 
-  describe('mock()', () => {
+  describe('mockResponse()', () => {
     it('should respond to mocked json request', async () => {
       server = await testServer();
       server.mockResponse('/api/foo', { body: { foo: 'foo' } });
@@ -189,6 +189,18 @@ describe('testServer', () => {
       const res = await fetch('http://localhost:8080/foo');
       expect(res).to.exist;
       expect(res.status).to.equal(500);
+    });
+    it('should respect custom headers when responding to mocked file', async () => {
+      server = await testServer();
+      server.mockResponse('/foo', {
+        headers: { 'X-foo': 'foo', 'Cache-Control': 'private, max-age=2' },
+        body: './test/unit/fixtures/mock/test.json',
+      });
+      const response = await fetch('http://localhost:8080/foo');
+      expect(response.headers.get('Cache-Control')).to.equal(
+        'private, max-age=2',
+      );
+      expect(response.headers.get('X-Foo')).to.equal('foo');
     });
   });
 

